google-ads-examples/src/main/java/com/google/ads/googleads/examples/remarketing/UploadCallConversion.java

```diff
@@ -0,0 +1,166 @@
+      params.conversionValue = Double.parseDouble("INSERT_CONVERSION_VALUE");
+    }
+
+    GoogleAdsClient googleAdsClient;
+    try {
+      googleAdsClient = GoogleAdsClient.newBuilder().fromPropertiesFile().build();
+    } catch (FileNotFoundException fnfe) {
+      System.err.printf(
+          "Failed to load GoogleAdsClient configuration from file. Exception: %s%n", fnfe);
+      return;
+    } catch (IOException ioe) {
+      System.err.printf("Failed to create GoogleAdsClient. Exception: %s%n", ioe);
+      return;
+    }
+
+    try {
+      new UploadCallConversion()
+          .runExample(
+              googleAdsClient,
+              params.customerId,
+              params.conversionActionId,
+              params.callerId,
+              params.callStartDateTime,
+              params.conversionValue);
+    } catch (GoogleAdsException gae) {
+      // GoogleAdsException is the base class for most exceptions thrown by an API request.
+      // Instances of this exception have a message and a GoogleAdsFailure that contains a
+      // collection of GoogleAdsErrors that indicate the underlying causes of the
+      // GoogleAdsException.
+      System.err.printf(
+          "Request ID %s failed due to GoogleAdsException. Underlying errors:%n",
+          gae.getRequestId());
+      int i = 0;
+      for (GoogleAdsError googleAdsError : gae.getGoogleAdsFailure().getErrorsList()) {
+        System.err.printf("  Error %d: %s%n", i++, googleAdsError);
+      }
+    }
+  }
+
+  /**
+   * Runs the example.
+   *
+   * @param googleAdsClient the client.
+   * @param customerId the customer ID.
+   * @param conversionActionId the conversion action ID.
+   * @param callerId the caller ID.
+   * @param callStartDateTime the call start date time
+   * @param conversionValue the value of the conversion in USD.
+   */
+  private void runExample(
+      GoogleAdsClient googleAdsClient,
+      long customerId,
+      String conversionActionId,
+      String callerId,
+      String callStartDateTime,
+      double conversionValue) {
+    // Create a call conversion by specifying currency as USD.
+    CallConversion conversion =
+        CallConversion.newBuilder()
+            .setConversionAction(StringValue.of(conversionActionId))
+            .setCallerId(StringValue.of(callerId))
+            .setCallStartDateTime(StringValue.of(callStartDateTime))
+            .setConversionValue(DoubleValue.of(conversionValue))
+            .setCurrencyCode(StringValue.of("USD"))
+            .build();
+
+    // Uploads the call conversion to the API.
+    try (ConversionUploadServiceClient conversionUploadServiceClient =
+        googleAdsClient.getLatestVersion().createConversionUploadServiceClient()) {
+      // Partial failure MUST be enabled for this request.
+      UploadCallConversionsResponse response =
+          conversionUploadServiceClient.uploadCallConversions(
+              String.valueOf(customerId), ImmutableList.of(conversion), true, false);
```

Comment:
The last two booleans could be confusing. I think it's worthwhile to either:

a) Add comments before each one explaining which one's partial failure and which one's validate only
or
b) Construct a request object using a builder, which will be self-documenting via the setter methods you call

Reply:
I would suggest option (b) in case there are changes to flattened method signatures moving forward.